sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/security/HardcodedCredentialsCheck.java

```diff
@@ -72,7 +79,12 @@ private Set<String> credentialWordsSet() {
 
   @Override
   public void scanFile(XmlFile file) {
-    checkElements(file.getDocument());
+    isWebConfigFile = Xml.isDotNetApplicationConfig(file.getInputFile());
+    if (isWebConfigFile) {
+      evaluateAsList(WEB_CONFIG_CREDENTIALS_PATH, file.getDocument()).forEach(this::checkElements);
+    } else {
+      checkElements(file.getDocument());
+    }
     checkSpecialCases(file);
   }
```

Comment:
We could separate the web.config logic:
```
    if (Xml.isDotNetApplicationConfig(file.getInputFile())) {
      evaluateAsList(WEB_CONFIG_CREDENTIALS_PATH, file.getDocument()).stream()
        .filter(password -> !isValidWebConfigCredential( password.getNodeValue() ))
        .forEach(password -> reportIssue(password, String.format("\"%s\" detected here, make sure this is not a hard-coded credential.", password.getLocalName())));
    } else {
      checkElements(file.getDocument());
      checkSpecialCases(file);
    }
```
And find a way to extract a common `reportIssue` method.